### PR TITLE
Update Convex Example

### PR DESCRIPTION
**Pull request by ianmacartney**
Updates the convex example to use a more recent version, along with newer style syntax and best practices.

### Adding or Updating Examples

- The "examples guidelines" are followed from our contributing doc https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md
- Make sure the linting passes by running `pnpm build && pnpm lint`. See https://github.com/vercel/next.js/blob/canary/contributing/repository/linting.md

### CODE REVIEW

**Comment by ianmacartney** [comment]
I can't get info from those errors: is it just b/c this is on my personal org, not owned by next.js?

**Comment by leerob** [comment]
Looks like Prettier failed here 🙏

**Comment by ianmacartney** [comment]
Very strange. That failing file was only touched by the merge commit. Rebasing w/o that merge commit and running `pnpm lint` passes without changing anything. ¯\_(ツ)_/¯

File: examples/convex/README.md [hunk 1]
```diff
@@ -20,14 +20,14 @@ yarn create next-app --example convex convex-app
...
 
-Run
+While developing:
 
 ```bash
 npm run dev
 ```
 
-to run `next dev` and a Convex file watcher at the same time. This command will log you into Convex, so you'll need to create a Convex account if this is your first project.
+This runs `next dev` and `convex dev` at the same time. This command will log you into Convex, so you'll need to create a Convex account if this is your first project.
```
Comment (leerob) [inline examples/convex/README.md:29]:
```suggestion
This command runs `next dev` and `convex dev` at the same time. This command will log you into Convex, so you'll need to create a Convex account if this is your first project.
```

**Pull request merged**